_TypeRecord constructor shadow warning with GCC 10

## Bug Report

### Build Problem

[We are seeing](https://github.com/olive-editor/olive/issues/1286) shadow warnings from [src/opentimelineio/typeRegistry.h](https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/68915ea6c393e9455e612c64a5453404d5a6a070/src/opentimelineio/typeRegistry.h#L111) with GCC 10. With warnings as errors this breaks compilation.

## To Reproduce

1. **Operating System:** Ubuntu 20.10
2. **Python version:** (not relevant)
3. **Example snippet** that demonstrates the issue - if it's a build issue:
    (build Olive, which uses OTIO in [saveotio.cpp](https://github.com/olive-editor/olive/blob/master/app/task/project/saveotio/saveotio.cpp))
4. **OpenTimelineIO release version or commit hash:** 68915ea (latest)
5. **Compiler information:** gcc version 10.2.0 (Ubuntu 10.2.0-13ubuntu1) 
    <details><summary>Full info</summary><pre><code>
    Using built-in specs.
    COLLECT_GCC=gcc
    COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/10/lto-wrapper
    OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa:hsa
    OFFLOAD_TARGET_DEFAULT=1
    Target: x86_64-linux-gnu
    Configured with: ../src/configure -v --with-pkgversion='Ubuntu 10.2.0-13ubuntu1' --with-bugurl=file:///usr/share/doc/gcc-10/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-10 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-10-JvwpWM/gcc-10-10.2.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-10-JvwpWM/gcc-10-10.2.0/debian/tmp-gcn/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
    Thread model: posix
    Supported LTO compression algorithms: zlib zstd
    gcc version 10.2.0 (Ubuntu 10.2.0-13ubuntu1) 
    user@users-inspiron:/opt/breakpad$ g++ -v
    Using built-in specs.
    COLLECT_GCC=g++
    COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/10/lto-wrapper
    OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa:hsa
    OFFLOAD_TARGET_DEFAULT=1
    Target: x86_64-linux-gnu
    Configured with: ../src/configure -v --with-pkgversion='Ubuntu 10.2.0-13ubuntu1' --with-bugurl=file:///usr/share/doc/gcc-10/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-10 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-10-JvwpWM/gcc-10-10.2.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-10-JvwpWM/gcc-10-10.2.0/debian/tmp-gcn/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
    Thread model: posix
    Supported LTO compression algorithms: zlib zstd
    gcc version 10.2.0 (Ubuntu 10.2.0-13ubuntu1) 
    user@users-inspiron:/opt/breakpad$ gcc-10 -v
    Using built-in specs.
    COLLECT_GCC=gcc-10
    COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/10/lto-wrapper
    OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa:hsa
    OFFLOAD_TARGET_DEFAULT=1
    Target: x86_64-linux-gnu
    Configured with: ../src/configure -v --with-pkgversion='Ubuntu 10.2.0-13ubuntu1' --with-bugurl=file:///usr/share/doc/gcc-10/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-10 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-10-JvwpWM/gcc-10-10.2.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-10-JvwpWM/gcc-10-10.2.0/debian/tmp-gcn/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
    Thread model: posix
    Supported LTO compression algorithms: zlib zstd
    gcc version 10.2.0 (Ubuntu 10.2.0-13ubuntu1) 
    user@users-inspiron:/opt/breakpad$ g++-10 -v
    Using built-in specs.
    COLLECT_GCC=g++-10
    COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/10/lto-wrapper
    OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa:hsa
    OFFLOAD_TARGET_DEFAULT=1
    Target: x86_64-linux-gnu
    Configured with: ../src/configure -v --with-pkgversion='Ubuntu 10.2.0-13ubuntu1' --with-bugurl=file:///usr/share/doc/gcc-10/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-10 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-10-JvwpWM/gcc-10-10.2.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-10-JvwpWM/gcc-10-10.2.0/debian/tmp-gcn/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
    Thread model: posix
    Supported LTO compression algorithms: zlib zstd
    gcc version 10.2.0 (Ubuntu 10.2.0-13ubuntu1)
    </code></pre></details>

## Expected Behavior

No shadow warnings, preferably without resorting to another compiler or compiler version.

## Logs

<details><summary>Compile log excerpt</summary><pre><code>
In file included from /opt/OpenTimelineIO/build/include/opentimelineio/serializableObject.h:8,
                 from /opt/OpenTimelineIO/build/include/opentimelineio/serializableObjectWithMetadata.h:4,
                 from /opt/OpenTimelineIO/build/include/opentimelineio/timeline.h:4,
                 from /opt/olive/source/app/task/project/saveotio/saveotio.h:24,
                 from /opt/olive/source/app/core.cpp:59:
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h: In constructor ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord::_TypeRecord(std::string, int, std::string, std::function<opentimelineio::v1_0::SerializableObject*()>)’:
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:111:83: error: declaration of ‘create’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  111 |                     std::string class_name, std::function<SerializableObject* ()> create) {
      |                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:106:47: note: shadowed declaration is here
  106 |         std::function<SerializableObject* ()> create;
      |                                               ^~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:111:33: error: declaration of ‘class_name’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  111 |                     std::string class_name, std::function<SerializableObject* ()> create) {
      |                     ~~~~~~~~~~~~^~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:105:21: note: shadowed declaration is here
  105 |         std::string class_name;
      |                     ^~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:110:50: error: declaration of ‘schema_version’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  110 |         _TypeRecord(std::string schema_name, int schema_version,
      |                                              ~~~~^~~~~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:104:13: note: shadowed declaration is here
  104 |         int schema_version;
      |             ^~~~~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:110:33: error: declaration of ‘schema_name’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  110 |         _TypeRecord(std::string schema_name, int schema_version,
      |                     ~~~~~~~~~~~~^~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:103:21: note: shadowed declaration is here
  103 |         std::string schema_name;
      |                     ^~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h: In constructor ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord::_TypeRecord(std::string, int, std::string, std::function<opentimelineio::v1_0::SerializableObject*()>)’:
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:111:83: error: declaration of ‘create’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  111 |                     std::string class_name, std::function<SerializableObject* ()> create) {
      |                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:106:47: note: shadowed declaration is here
  106 |         std::function<SerializableObject* ()> create;
      |                                               ^~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:111:33: error: declaration of ‘class_name’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  111 |                     std::string class_name, std::function<SerializableObject* ()> create) {
      |                     ~~~~~~~~~~~~^~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:105:21: note: shadowed declaration is here
  105 |         std::string class_name;
      |                     ^~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:110:50: error: declaration of ‘schema_version’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  110 |         _TypeRecord(std::string schema_name, int schema_version,
      |                                              ~~~~^~~~~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:104:13: note: shadowed declaration is here
  104 |         int schema_version;
      |             ^~~~~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:110:33: error: declaration of ‘schema_name’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  110 |         _TypeRecord(std::string schema_name, int schema_version,
      |                     ~~~~~~~~~~~~^~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:103:21: note: shadowed declaration is here
  103 |         std::string schema_name;
      |                     ^~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h: In constructor ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord::_TypeRecord(std::string, int, std::string, std::function<opentimelineio::v1_0::SerializableObject*()>)’:
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:111:83: error: declaration of ‘create’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  111 |                     std::string class_name, std::function<SerializableObject* ()> create) {
      |                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:106:47: note: shadowed declaration is here
  106 |         std::function<SerializableObject* ()> create;
      |                                               ^~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:111:33: error: declaration of ‘class_name’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  111 |                     std::string class_name, std::function<SerializableObject* ()> create) {
      |                     ~~~~~~~~~~~~^~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:105:21: note: shadowed declaration is here
  105 |         std::string class_name;
      |                     ^~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:110:50: error: declaration of ‘schema_version’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  110 |         _TypeRecord(std::string schema_name, int schema_version,
      |                                              ~~~~^~~~~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:104:13: note: shadowed declaration is here
  104 |         int schema_version;
      |             ^~~~~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:110:33: error: declaration of ‘schema_name’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  110 |         _TypeRecord(std::string schema_name, int schema_version,
      |                     ~~~~~~~~~~~~^~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:103:21: note: shadowed declaration is here
  103 |         std::string schema_name;
      |                     ^~~~~~~~~~~
</code></pre></details>

## Additional Context

Might this be just a matter of renaming the constructor arguments (e.g. a `_` prefix)?

```cpp
    class _TypeRecord {
        std::string schema_name;
        int schema_version;
        std::string class_name;
        std::function<SerializableObject* ()> create;
        
        std::map<int, std::function<void (AnyDictionary*)>> upgrade_functions;
        
        _TypeRecord(std::string _schema_name, int _schema_version,
                    std::string _class_name, std::function<SerializableObject* ()> _create) {
            this->schema_name = _schema_name;
            this->schema_version = _schema_version;
            this->class_name = _class_name;
            this->create = _create;
        }
```

This warning is, in my opinion, an annoyance. 

```
class Foo {
   int bar;
public:
   Foo(int bar) : bar(bar) {}
```

is well defined and correct code that properly initializes bar.

Had to get my opinion in about gcc being an overbearing and overprotective nanny, and not liking warnings that don't highlight problems, but someone's opinion, grrr.
-----------------
Okay, now that that's out the way ;) My second opinion is that given the choice to suppress the warning with a flag, or add an underbar, I'd prefer the underbar, because we shouldn't be imposing opinions about warnings on other projects' builds.

Please feel free to post a PR to address this warning.

Thanks for the report!
-----------------

For reference, here is the specific warning from gcc:

```
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h: In constructor ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord::_TypeRecord(std::string, int, std::string, std::function<opentimelineio::v1_0::SerializableObject*()>)’:
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:111:83: error: declaration of ‘create’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  111 |                     std::string class_name, std::function<SerializableObject* ()> create) {
      |                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:106:47: note: shadowed declaration is here
  106 |         std::function<SerializableObject* ()> create;
      |                                               ^~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:111:33: error: declaration of ‘class_name’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  111 |                     std::string class_name, std::function<SerializableObject* ()> create) {
      |                     ~~~~~~~~~~~~^~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:105:21: note: shadowed declaration is here
  105 |         std::string class_name;
      |                     ^~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:110:50: error: declaration of ‘schema_version’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  110 |         _TypeRecord(std::string schema_name, int schema_version,
      |                                              ~~~~^~~~~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:104:13: note: shadowed declaration is here
  104 |         int schema_version;
      |             ^~~~~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:110:33: error: declaration of ‘schema_name’ shadows a member of ‘opentimelineio::v1_0::TypeRegistry::_TypeRecord’ [-Werror=shadow]
  110 |         _TypeRecord(std::string schema_name, int schema_version,
      |                     ~~~~~~~~~~~~^~~~~~~~~~~
/opt/OpenTimelineIO/build/include/opentimelineio/typeRegistry.h:103:21: note: shadowed declaration is here
  103 |         std::string schema_name;
      |                     ^~~~~~~~~~~
```

--- a/src/opentimelineio/typeRegistry.h
+++ b/src/opentimelineio/typeRegistry.h
@@ -107,12 +107,12 @@ private:
         
         std::map<int, std::function<void (AnyDictionary*)>> upgrade_functions;
         
-        _TypeRecord(std::string schema_name, int schema_version,
-                    std::string class_name, std::function<SerializableObject* ()> create) {
-            this->schema_name = schema_name;
-            this->schema_version = schema_version;
-            this->class_name = class_name;
-            this->create = create;
+        _TypeRecord(std::string _schema_name, int _schema_version,
+                    std::string _class_name, std::function<SerializableObject* ()> _create) {
+            this->schema_name = _schema_name;
+            this->schema_version = _schema_version;
+            this->class_name = _class_name;
+            this->create = _create;
         }
         
         SerializableObject* create_object() const;

